sdk/core/Azure.Core/samples/Diagnostics.md

```diff
@@ -176,6 +179,117 @@ Most of Azure SDKs released in or after November 2023 have OpenTelemetry support
+
+### Avoiding double-collection of HTTP activities
+
+_Note: if you use Azure Monitor Distro for ASP.NET Core, you may skip this section. The distro configures telemetry collection preventing duplication._
+
+Azure SDK traces all HTTP calls using `Azure.Core.Http` source. If you enable it along with generic HTTP client instrumentation such as `OpenTelemetry.Instrumentation.Http`, it would lead to double-collection of HTTP calls originating from Azure client libraries.
+
+Unlike generic HTTP activities, Azure SDK HTTP activities include Azure-specific attributes such as request identifiers usually passed to and from Azure serices in `x-ms-client-request-id`, `x-ms-request-id` or similar request and respose headers. This data may be important when correlating client and server telemetry or creating support tickets.
+
+To avoid double-collection you may either 
+- enrich generic HTTP client activities with Azure request identifiers and disable Azure SDK HTTP activities.
+- filter out duplicated generic HTTP client activities.
+
+#### Enriching generic HTTP activities with Azure request identifiers
+
+For example, you can do it with the following code snippet on .NET Core.
+
+```csharp
+// Messaging libraries still require feature flag.
+AppContext.SetSwitch("Azure.Experimental.EnableActivitySource", true)
+
+builder.Services.AddOpenTelemetry()
+    .WithTracing(tracerProviderBuilder => tracerProviderBuilder
+        .AddSource("Azure.Storage.*")
+        .AddSource("Azure.Messaging.*")
+        .AddHttpClientInstrumentation(o => {
+            o.EnrichWithHttpResponseMessage = (activity, response) =>
+            {
+                if (response.RequestMessage.Headers.TryGetValues("x-ms-client-request-id", out var clientRequestId))
+                {
+                    activity.SetTag("az.client_request_id", clientRequestId);
+                }
+                if (response.Headers.TryGetValues("x-ms-request-id", out var requestId))
+                {
+                    activity.SetTag("az.service_request_id", requestId);
+                }
+            };
+        })
+        ...)
+```
+
+Here we enabled tracing for storage and messaging groups and also enabled HTTP client instrumentation providing hooks to collect request id headers. It's recommended to use provided attribute names to stay consistent with Azure SDK behavior.
+
+_Note: request identifiers are also stamped on HTTP logs emitted by `Azure.Core`. If you collect such logs, or record request identifiers using other means, you might not need to record them on traces as well._
+
+#### Filtering out duplicated HTTP client activities
+
+Another approach would be to filter out generic HTTP client activities with the following approach:
+
+```csharp
+.WithTracing(tracerProviderBuilder => tracerProviderBuilder
+    .AddSource("Azure.*")
+    .AddHttpClientInstrumentation(o => {
+        o.EnrichWithHttpResponseMessage = (activity, response) =>
```

Comment:
meant to show [this](https://github.com/Azure/azure-sdk-for-net/blob/cbc04f8981f3fca3dd18d905ff9549480f17c3b6/sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/OpenTelemetryBuilderExtensions.cs#L149-L164) ?

Reply:
great catch,, not sure how this has happened